cmd/prometheus/main.go

```diff
@@ -118,7 +122,7 @@ func Main() int {
 		GoVersion: version.GoVersion,
 	}
 
-	webHandler := web.New(memStorage, queryEngine, targetManager, ruleManager, version, flags, &cfg.web)
+	webHandler := web.New(memStorage, queryEngine, queryCtx, targetManager, ruleManager, version, flags, &cfg.web)
```

Comment:
Yes, keeping it at one is better. A lot of options there. Probably a good idea to verify whether the constructor uses sane defaults where possible.